nep-12.mediawiki

```diff
@@ -0,0 +1,2130 @@
+  type: string; // `NO_PROVIDER`|`CONNECTION_DENIED`
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+getNetworks()
+.then(response => {
+  const {
+    networks,
+    defaultNetwork,
+  } = response.networks;
+
+  console.log('Networks: ' + networks);
+  // eg. ["MainNet", "TestNet", "PrivateNet"]
+
+  console.log('Default network: ' + defaultNetwork);
+  // eg. "MainNet"
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_DENIED:
+      console.log('The user rejected the request to connect with your dApp');
+      break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request, return:
+- The list of NEO networks that the wallet is able to connect to. The return values will be used by the dapp developer to communicate to which network they would like their request to be directed to. The wallet provider will be responsible for the logistics of which node to submit any request to for each network alias provided.
+- The wallet provider is to also return the default network that the wallet UI is currently set to by user.
+
+Example
+<pre>
+{
+  networks: ["MainNet", "TestNet", "PrivateNet"],
+  defaultNetwork: "TestNet",
+}
+</pre>
+
+
+====getAccount====
+Return the Account that is currently connected to the dApp.
+
+
+'''Method Interface'''
+
+<pre>
+function getAccount(): Promise<{ address: string, label: string }>
+</pre>
+
+
+'''Input arguments'''
+
+None
+
+
+'''Success return value'''
+
+<pre>
+interface Account {
+  address: string; // Address of the connected account
+  label?: string; // A label the users has set to identify their wallet
+}
+</pre>
+
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string; // `NO_PROVIDER`|`CONNECTION_DENIED`
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+getAccount()
+.then((account: Account) => {
+  const {
+    address,
+    label
+  } = account;
+
+  console.log('Provider address: ' + address);
+  console.log('Provider account label (Optional): ' + label);
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+    case CONNECTION_DENIED:
+      console.log('The user rejected the request to connect with your dApp');
+      break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request, return:
+- The address and label of account currently selected by the user in the wallet interface
+
+Example
+<pre>
+{
+  address: 'AeysVbKWiLSuSDhg7DTzUdDyYYKfgjojru',
+  label: 'My Spending Wallet'
+}
+</pre>
+
+
+====getPublicKey====
+Return the public key of the Account that is currently connected to the dApp.
+
+
+'''Method Interface'''
+
+<pre>
+function getPublicKey(): Promise<{ address: string, publicKey: string }>
```

Comment:
Is this by chance allowing multisig in the future? If so, an index would be welcome on function, so you can get pubkey 0, pubkey 1, etc, from the same Address.

Reply:
This is based on the single public key returned from `neon-js`.

https://github.com/CityOfZion/neon-js/blob/master/packages/neon-core/src/wallet/Account.ts#L209

Reply:
I get that.. but yet, user account mat be multisig I think. Many people wonder why multisig is not deeply used... but if we wont support and incentive it, no one will use it, thats sure.